### PR TITLE
Add Edit payout button

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

closes https://github.com/jbx-protocol/juice-interface/issues/785

A bunch of rework and have made the modal form a bit more performant. I have already uncovered some bugs through some quick testing, but I think they may already exist in the codebase, so if reviewers are willing, i'd prefer to create issues and fix after merge to avoid growing this PR any bigger/longer.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [ ] I have evaluated the
      [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines)
      for this PR.
- [ ] I have tested this PR in
      [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

**Comment by aeolianeth** [comment]
@wraeth-eth I've had a quick review of this but will need to go into greater depth later.

File: src/components/v2/V2Project/V2FundingCycleSection/PayoutSplitsCard.tsx [hunk 1]
```diff
@@ -134,6 +171,32 @@ export default function PayoutSplitsCard({
         onCancel={() => setDistributePayoutsModalVisible(false)}
         onConfirmed={() => window.location.reload()}
       />
+      <EditPayoutsModal
+        visible={editPayoutModalVisible}
+        confirmLoading={payoutsModalLoading}
+        onCancel={() => setEditPayoutModalVisible(false)}
+        onConfirm={async newSplits => {
```
[the remark this answers is not in the document]
Comment (wraeth-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/PayoutSplitsCard.tsx:178]: no particular reason - happy to reassess and move 👍

[the remark this answers is not in the document]
Comment (wraeth-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/PayoutSplitsCard.tsx:178]: I think my reasoing was to try and keep the tx as close to the original source of truth as possible, but can look at moving to contain in the modal (separation of concern, etc etc)

File: src/components/v2/V2Project/V2FundingCycleSection/modals/EditPayoutsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        confirmLoading={confirmLoading}
+        title="Edit payouts"
+        okText="Save payouts"
+        cancelText={confirmLoading ? 'Close' : 'Cancel'}
+        onOk={() => onSplitsConfirmed(editingSplits)}
+        onCancel={onCancel}
+        width={720}
+      >
+        <div>
+          <DescriptionParagraphOne />
```
[the remark this answers is not in the document]
Comment (wraeth-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/modals/EditPayoutsModal.tsx:167]: Yeah me too! :) Makes it a bit more human readable/scannable

**Comment by johnnyd-eth** [comment]
So good @wraeth-eth. My main comment is that I think we need to make it clear somehow why we can't adjust payouts by amounts in this modal (because distribution limit isn't set in this tx to save gas). 

- It's purely a way to edit percentages, so I'm wondering if we should even rename the button from 'Edit payouts' to 'Edit percentages' or something.  
- The modal description needs some info about how amounts can't be set in this modal because the transaction doesn't handle distribution limit, and that if they want to adjust payouts as amounts (adjust the overall distribution limit) they'll need to use the reconfigure transaction (possibly make a link to open the reconfigure funding cycle modal)

- Can we show amounts in these cards (if limit is not infinite)?
Screen Shot 2022-06-12 at 12 45 01 pm

- When the percentages don't add up to 100%, I think we need to clearly indicate how much remainder will be allocated to the owner. We should match the extra hard-coded card we've got in the create flow for this. In this case we can remove the 'By default, all unallocated funds can be distributed to the project owner's wallet.' sentence from the description as it will be covered in the info tooltip of the remainder card. Also, will need to show a different message in this tooltip depending on if limit is infinite or specific

Screen Shot 2022-06-12 at 12 58 12 pm
Screen Shot 2022-06-12 at 1 09 22 pm

**Comment by wraeth-eth** [comment]
Thanks for feedback @johnnyd-eth - have made some comments on discord

Have also fixed up a couple of suggestions from you.

Re Modal needs the remainder split card (like in create/reconfig), not quite sure I follow. Do you mean the owner split card? If so, this is what it looks like:
image

**Review by johnnyd-eth — APPROVED** [review]
LFG

**Pull request merged**